typescript/sdk/src/hook/EvmHookModule.ts

```diff
@@ -184,29 +287,296 @@ export class EvmHookModule extends HyperlaneModule<
+  }): Promise<AnnotatedEV5Transaction[]> {
+    const gasParamsToSet: InterchainGasPaymaster.GasParamStruct[] = [];
+    for (const [remote, gasOverhead] of Object.entries(targetOverheads)) {
+      // Note: non-EVM remotes actually *are* supported, provided that the remote domain is in the MultiProvider.
+      // Previously would check core metadata for non EVMs and fallback to multiprovider for custom EVMs
+      const remoteDomain = this.multiProvider.tryGetDomainId(remote);
+
+      if (!remoteDomain) {
+        this.logger.warn(
+          `Skipping overhead ${this.chain} -> ${remote}. Expected if the remote domain is not in the MultiProvider.`,
+        );
+        continue;
+      }
+
+      // only update if the gas overhead has changed
+      if (currentOverheads?.[remote] !== gasOverhead) {
+        this.logger.debug(
+          `Setting gas params for ${this.chain} -> ${remote}: gasOverhead = ${gasOverhead} gasOracle = ${gasOracle}`,
+        );
+        gasParamsToSet.push({
+          remoteDomain,
+          config: {
+            gasOverhead,
+            gasOracle,
+          },
+        });
+      }
+    }
+
+    if (gasParamsToSet.length === 0) {
+      return [];
+    }
+
+    return [
+      {
+        annotation: `Updating overhead for domains ${Object.keys(
+          targetOverheads,
+        ).join(', ')}...`,
+        chainId: this.domainId,
+        to: interchainGasPaymaster,
+        data: InterchainGasPaymaster__factory.createInterface().encodeFunctionData(
+          'setDestinationGasConfigs((uint32,(address,uint96))[])',
+          [gasParamsToSet],
+        ),
+      },
+    ];
+  }
+
+  protected async updateStorageGasOracle({
+    gasOracle,
+    currentOracleConfig,
+    targetOracleConfig,
+    targetOverhead,
+  }: {
+    gasOracle: Address;
+    currentOracleConfig?: IgpConfig['oracleConfig'];
+    targetOracleConfig: IgpConfig['oracleConfig'];
+    targetOverhead: IgpConfig['overhead'];
+  }): Promise<AnnotatedEV5Transaction[]> {
+    this.logger.info(`Updating gas oracle configuration from ${this.chain}...`);
+    const configsToSet: Array<StorageGasOracle.RemoteGasDataConfigStruct> = [];
+
+    for (const [remote, target] of Object.entries(targetOracleConfig)) {
+      // Note: non-EVM remotes actually *are* supported, provided that the remote domain is in the MultiProvider.
+      // Previously would check core metadata for non EVMs and fallback to multiprovider for custom EVMs
+      const current = currentOracleConfig?.[remote];
+      const remoteDomain = this.multiProvider.tryGetDomainId(remote);
+
+      if (!remoteDomain) {
+        this.logger.warn(
+          `Skipping gas oracle update ${this.chain} -> ${remote}. Expected if the remote domain is not in the MultiProvider.`,
+        );
+        continue;
+      }
+
+      // only update if the oracle config has changed
+      if (!current || !configDeepEquals(current, target)) {
+        configsToSet.push({ remoteDomain, ...target });
```

Comment:
There's no validation in these Config schemas. This might lead to weird unexpected behavior. For example,

If `target` has `remoteDomain` key, it will overwrite the fetched `remoteDomain`. Might be intentional, but it's also not a valid `StorageGasOracleConfigSchema`. I believe its possible to have something like 
```typescript
const target: StorageGasOracleConfigSchema = { 
  gasPrice: "1000", 
  tokenExchangeRate: "1000", 
  blablablabla: 7777 
}
```

Consider using Schema.parse to 1) validate config schemas, and 2) filter out extra parameters

I probably have to do this for my PRs too ;)

Reply:
i'm aware it's not great in ts land with objects so i tried to be mindful about only selecting/using fields i require from an object - but I'll add schema parsing at the top of constructor/deploy/update functions in both ISM + Hook modules as it never hurts to be safe!